src/wp-includes/http.php

```diff
@@ -589,6 +589,15 @@ function wp_http_validate_url( $url ) {
 	$host        = trim( $parsed_url['host'], '.' );
 
 	if ( ! $same_host ) {
+		if (
+			function_exists( 'filter_var' )
+			&& defined( 'FILTER_VALIDATE_DOMAIN' )
+			&& defined( 'FILTER_FLAG_HOSTNAME' )
+			&& ! filter_var( $host, FILTER_VALIDATE_IP )
```

Comment:
There's no constant check for `FILTER_VALIDATE_IP`, but I think we probably only need the `filter_var()` function exists check. And even instead of even that it could just check `extension_loaded('filter')`